### ISSUE TITLE
Remove Neve dependency from TPC

### ISSUE THREAD

**Issue by preda-bogdan** [Codeinwp/templates-patterns-collection#154]
### Description:
Remove Neve dependency from Neve to allow TPC to work as a stand-alone.
1. We would keep the current requirements of Template Cloud (it will require Neve Pro) to give access to saving and managing user templates
2. We would allow Starter Sites to install Neve if the theme is not present.

**Comment by preda-bogdan** [comment]
@selul @cristian-ungureanu 

Right now when we have TPC as a stand-alone we are presented with a notice to install Neve: 
![image](https://user-images.githubusercontent.com/23024731/178972657-a2fb5d07-478d-46dd-892b-5b33d1bbbcf3.png)

And when importing a template a pop-up is presented:
![image](https://user-images.githubusercontent.com/23024731/178972745-6f5ba762-c20a-4fd6-bee3-929bff5dc378.png)

Is the objective to install and activate Neve without presenting these?

**Comment by preda-bogdan** [comment]
Thank you for the feedback.

So here is my action plan:
1. Remove the Notice that Neve is required from Plugins
2. I will remove the pop-up when Neve is not present on Starter Sites and add Neve as part of the required Install.
3. For Page Templates I will let them import the template without requiring them to have Neve.
4. I will ensure that if the Neve Pro license is available, the Template Cloud is available even if the Neve theme is not being used. My Library should be available on other themes if the License is active.

For the second phase, I will open a new issue where we can go into details after we release these changes.

**Comment by preda-bogdan** [comment]
> > I will ensure that if the Neve Pro license is available, the Template Cloud is available even if the Neve theme is not being
> 
> here we can keep the current upsell for PRO

Sure, it will remain the same as it is now, only ensuring that it can be used if an existing license is present.

**Comment by preda-bogdan** [comment]
So I implemented the changes as discussed.
How it works now:
1. You can have only the TPC plugin installed on a fresh install and it will allow you to import Starter Sites
    Here if Neve is not present it will install it automatically and show you this during the Import process
    This also applies to individual pages of a Starter Site
2. Page Templates are allowed to be imported without requiring or installing Neve
    However, here I think we should also make sure that the templates will display correctly regardless of the theme used.
3. I am using the License of Neve Pro so if the plugin is active and the License is valid access to Template Cloud and other Pro features is allowed. This should work without having Neve active. If the plugin or license is not present the same upsell is present as before.
    I think here in the future it would benefit to have its own licensing mechanism and maybe migrate the license from Neve.


Let me know your feedback on this and/or if there are things that I did not cover. Thank you!

**Comment by cristian-ungureanu** [comment]
Hey @preda-bogdan, overall I think we are on the right path. I have some mentions and also some questions.
We have this scenario: TPC with the TwentyTwenty-One theme.

1. There is still a notice saying you should buy Neve Pro, business plan or higher with a link to the upgrade page. Should we keep it / replace it / remove it?
<img width="1372" alt="Screenshot 2022-07-21 at 14 09 04" src="https://user-images.githubusercontent.com/9929553/180205152-d712b026-ea67-4904-9c1c-f883d66e2d7c.png">

2. When I want to import something, the requirement of the Neve theme is displayed within the settings section. Maybe a separate section named "Required theme" might be more suitable.
<img width="704" alt="Screenshot 2022-07-21 at 14 42 19" src="https://user-images.githubusercontent.com/9929553/180206089-5c67d423-d1bf-421f-93f4-6eb168942c16.png">

3. The modal seems like a choice to me. If I want to import the template and I agree with all the settings that it will make, I click on "Import entire site". If not, I have the option to click on "I want to import just the templates". Clicking on that installs and activates Neve automatically without saying that it will do so. Maybe we should add here in the modal the text "will install Otter and Neve theme"
<img width="728" alt="Screenshot 2022-07-21 at 14 51 19" src="https://user-images.githubusercontent.com/9929553/180207389-b2e57b21-8062-4249-9991-60da9bb6ce3c.png">

4. Page templates should definitely be improved.

5. The main pain point in this scenario is to use the Templates Cloud feature. If you have Astra and you go to My Library tab, you will be asked to get Neve Pro.
<img width="1919" alt="Screenshot 2022-07-21 at 15 00 09" src="https://user-images.githubusercontent.com/9929553/180210963-62ae8c35-93f6-43dd-af92-cd362770b399.png">
Let's say someone gets neve pro just for this and installs it. Then they'll find out that Neve Pro will only work with Neve theme. So what they will need to do is to install Neve and Neve Pro, activate the license, then get back to Astra as that was their main theme.
Basically, right now, if you want to keep your theme that's other than Neve you could only import page templates ( that are not working well with other themes).

**Comment by preda-bogdan** [comment]
Thank you for the feedback I agree with points 2. and 3. we can make it more obvious that the theme will be installed and activated.

For point 1. I think we can keep it for now and just change the upsell if we proceed to have TPC as a self-licensed product.

For 4. I would create a separate issue where we can review the Page Templates and change them to work independent of the theme used. What do you guys think?

What do you recommend for 5.? I agree that the current implementation is not optimal, how would you tackle using the Neve Pro license without requiring to prior have Neve installed? We would require the same mechanism as we used on Sparks or Otter and have a self-license, but as mentioned above this is part of the second phase from my understanding.

Let me know and I will implement the required changes.
Thank you!

**Comment by cristian-ungureanu** [comment]
A solution for 5 could be to replace the notice. If Neve is not installed we can say "Coming soon. Right now this feature is not available with your current setup. if you want to use it, you need to install Neve + Neve Pro", and keep the link to the upgrade page. If Neve is installed but no Neve pro we could show the current notice.

For 4 I agree this could be handled in another issue.

Let's wait a bit to see what @selul thinks about it.

**Comment by selul** [comment]
Hi guys, 

Nice work!  

For 1, I agree with Bogdan, it's just an intermediate phase, we can evaluate this later if you want. If is adding too much confusion we can hide entirely the starter sites / templates tab if neve is not active for now and just have my templates.

2/3/4 depends on 1, if we go on having on only with Neve, this is no longer an issue.

5. I really like Cristi's idea and the change in notice content. Just to make sure we have it like this if Neve is not on.

What do you think?

**Comment by preda-bogdan** [comment]
Thank you for the feedback!

I would keep the Starter Sites and better highlight the fact that we are also installing Neve if not present. This way we can still funnel some users to Neve.

I would hide the Page Templates if Neve is not installed until we make sure that the templates would work regardless of the theme present.

We can change the notice on Template Cloud if Neve is present or not as Chris suggested and later will put in place a license workflow.

Let me know if you guys think this would work for the first phase?

I will implement this at once and aim for a release next week.

**Comment by cristian-ungureanu** [comment]
@preda-bogdan this sounds good. Let's also keep the Page Templates tab with a "Coming soon" and an upsell to Neve if they want to use it right now.

Everything else sounds good 👍

**Comment by preda-bogdan** [comment]
@cristian-ungureanu I've updated the PR and added the changes discussed here.

Let me know if you have more feedback or if there are other things I should change: https://github.com/Codeinwp/templates-patterns-collection/pull/155

**Comment by cristian-ungureanu** [comment]
@preda-bogdan I had a look over this and I want to add some small changes:

<img width="792" alt="Screenshot 2022-07-27 at 17 27 06" src="https://user-images.githubusercontent.com/9929553/181273153-71bd5607-340c-4464-9d75-ad80f397932e.png">

- [ ] For this notice let's replace **Neve + Neve Pro** with **Neve theme and Neve Pro Addon plugin.**. I would add a link over Neve ( maybe to the wp org repo ) and another one over Neve Pro and show "Upgrade to Pro" button only if Neve is installed.

- [ ] Same for "My Library" notice. Let's also replace the title "Templates Cloud is a PRO Feature" with "Coming soon" if Neve theme is not installed.

---

<img width="700" alt="Screenshot 2022-07-27 at 17 39 35" src="https://user-images.githubusercontent.com/9929553/181276287-b26375c9-b90c-478f-ae8a-449f0880095c.png">

Let's replace the "Install and Activate Theme" with "Installing and Activating the Theme"

---

Not a big deal and also not introduced now but the content is shifting between the tabs.

https://vertis.d.pr/AoNnYn

**Comment by preda-bogdan** [comment]
@cristian-ungureanu Thank you, I've implemented the changes and updated the texts.
![image](https://user-images.githubusercontent.com/23024731/181291581-201dac8a-830a-46f4-b6bb-17be7ff5c9a5.png)

**Comment by preda-bogdan** [comment]
This is now released.
I've opened a new ticket here: https://github.com/Codeinwp/templates-patterns-collection/issues/161
to discuss and track progress on phase 2.

I'll close this and we can continue there.
